File: Maths/CountPrime.cxx
// Count prime numbers
// https://edabit.com/challenge/Ba4vZuzC7b2Zr6SPs

#include <bits/stdc++.h>
using namespace std;

class Solution{
	public:
	static bool is_prime(long long n){
		if(n==2)
			return true;
		for(auto i = 2; i <= sqrt(n); ++i){
			if(n%i==0)
				return false;
		}
		return true;
	}
	static int primeNumber(long long n){
		auto count = 0;		
		for(auto x = 2; x <= n; ++x){
			if(is_prime(x))
				count++;
		}
		return count;
	}
};

int main(){
	ios_base::sync_with_stdio(0);
	int n = 30;
	cout << Solution::primeNumber(n) << endl;
	return 0;
}

File: Maths/CountPrime.py
# Count prime numbers
# https://edabit.com/challenge/Ba4vZuzC7b2Zr6SPs

import math

def is_prime(n:int)->bool:
	if(n==2):
		return True
	i=2
	while(i<=(int)(math.sqrt(n))):
		if(n%i==0):
			return False
		i+=1
	return True

def primeNumber(n:int)->int:
	ct=0
	for x in range(2,n+1):
		if(is_prime(x)):
			ct+=1
	return ct


if __name__=='__main__':
	n = 30
	print(primeNumber(n),end="\n") # 10


